### PR TITLE
Add Medtronic serial number field on device selection

### PR DESCRIPTION
**Pull request by hntrdglss**
This PR adds a serial number field for direct Medtronic device uploads. The serial number field should require no more or fewer than six digits, and should ONLY allow numeric input.

It also adds new metrics when clicking on "Choose devices" on the user menu and when the "Done" button is clicked on the device selection page.

See  [Trello card](https://trello.com/c/TiTc17UL/) for more info.

@krystophv: mind taking a look when you have time? @gniezen reviewed and merged in his branch from the previous PR (#340), so it'd be nice to have another set of eyes.

### CODE REVIEW

File: lib/components/Upload.js [hunk 1]
```diff
@@ -119,6 +137,32 @@ export default class Upload extends Component {
     }
   }
 
+  onMedtronicSerialNumberInputChange() {
+    const { refs } = this;
+    let serial_number = refs.medtronic_serial_number;
```
Comment (krystophv) [inline lib/components/Upload.js]: I think I would prefer in this instance not to rely on grabbing the value from `refs` when it's being passed into the function through the event (via `event.target.value`). Unless there's something about the `ref` pattern here that I'm missing.

File: lib/components/Upload.js [hunk 2]
```diff
@@ -293,6 +343,28 @@ export default class Upload extends Component {
...
+    if (_.get(upload, 'source.driverId', null) !== 'Medtronic') {
+      return null;
+    }
+
+    return (
+      <div>
+        <div className={styles.textInputWrapper}>
+          <p>Enter your 6 digit serial number found on the back of your pump.</p>
+          <input
+            type="number"
```
Comment (krystophv) [inline lib/components/Upload.js]: A strange side-effect of using the `number` input type to limit input on this field is that 'e' is actually a valid input due to scientific notation. Not sure if it'll come up in the field, but we might need to do some additional filtering/validation on the handler for the input update.

File: lib/components/Upload.js (hunk 1 above)
Comment (hntrdglss) [inline lib/components/Upload.js]: No specific reason, that's just how the CareLink inputs are handled, so I used the same for consistency.

File: lib/components/Upload.js (hunk 2 above)
Comment (hntrdglss) [inline lib/components/Upload.js]: Ooh, good catch. Probably unlikely as you mentioned, but I can definitely add something to `onMedtronicSerialNumberInputChange`.

Comment (hntrdglss) [inline lib/components/Upload.js]: So tricky! `input type="number"` doesn't emit an `onChange` event when the input is `123e` until something comes AFTER the `e` (like `123e4`), so it was still allowing the `e` to be entered. I switched the `input` type to `text`, to resolve that.

**Comment by krystophv** [comment]
The fact that it doesn't emit a change event until something comes after the 'e' is pretty incredible (and actually sounds like a bug). The solution seems pretty robust. LGTM.

**Pull request merged**